Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+                }
+            }
+        }
+
+        public void SubmitAnalytics(GeneratorExecutionContext context)
+        {
+            var payload = string.Empty;
+
+            if (Environment.GetEnvironmentVariable("REALM_DISABLE_ANALYTICS") != null ||
+                Environment.GetEnvironmentVariable("CI") != null)
+            {
+                return;
+            }
+
+            _realmEnvMetrics[SdkFeature.UserId] = AnonymizedUserID;
+
+            ComputeHostOSNameAndVersion(out var osName, out var osVersion);
+
+            // TODO andrea: check if searching strings in this way works on non Windows platforms
+            switch (osName)
+            {
+                case string name when name.Contains(Metric.OperatingSystem.Windows):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.Windows;
+                    break;
+                case string name when name.Contains(Metric.OperatingSystem.MacOS):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.MacOS;
+                    break;
+                case string name when name.Contains(Metric.OperatingSystem.Linux):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.Linux;
+                    break;
+                default:
+                    Diagnostics.Add(
+                        SourceGenerator.Diagnostics.AnalyticsDebugInfo($"{osName} is not an operating system that we recognize."));
+                    break;
+            }
+
+            _realmEnvMetrics[SdkFeature.HostOsVersion] = osVersion;
+
+            bool prettyJson = false;
+#if DEBUG
+            prettyJson = true;
+#endif
+            payload = GetJsonPayload(prettyJson);
+
+#if !DEBUG
+            var base64Payload = Convert.ToBase64String(Encoding.UTF8.GetBytes(payload));
+
+            // TODO andrea: this has never been tested yet
+            SendRequest(
+                "https://data.mongodb-api.com/app/realmsdkmetrics-zmhtm/endpoint/metric_webhook/metric_stage?data=",
+                base64Payload,
+                string.Empty);
+#endif
+
+            if (Environment.GetEnvironmentVariable("REALM_PRINT_ANALYTICS") != null &&
+                Environment.GetEnvironmentVariable("REALM_DISABLE_ANALYTICS") == null)
+            {
+                // TODO andrea: temporarily reusing the common diagnostic for analytics, this likely needs to have its own
+                Diagnostics.Add(SourceGenerator.Diagnostics.AnalyticsDebugInfo($@"
+----------------------------------
+Analytics payload
+{payload}
+----------------------------------"));
+            }
+
+            // TODO andrea: we may actually look for something like INTERNAL_DEBUG, so that the user never sees this
+#if DEBUG
+            foreach (var diagnostic in Diagnostics)
+            {
+                // TODO andrea: this does not up show anywhere. Investigate why.
+                context.ReportDiagnostic(diagnostic);
```

Comment:
We talked about this... maybe it's because you're not on the main source generator thread? Just an idea

Reply:
This is about SGs. We've now moved this to the weaver.